### PR TITLE
remove d2d copies

### PR DESCRIPTION
**Pull request by ksivaman**
Signed-off-by: Kirthi Shankar Sivamani <ksivamani@nvidia.com>

### CODE REVIEW

**Comment by ksivaman** [comment]
@ptrendx Turns out the `save_for_backward` optimizations are not necessary since the memory consumed by it is already minimized. Any additional tensors we pass only for the FP8 or gradient accumulation fusion cases are free because they are a part of the forward arguments.

File: transformer_engine/pytorch/csrc/ts_fp8_op.cpp [hunk 1]
```diff
@@ -92,11 +92,11 @@ at::Tensor te_gemm_ts(at::Tensor A,
   bool accumulate_arg = static_cast<bool>(accumulate);
   bool use_split_accumulator_arg = static_cast<bool>(use_split_accumulator);
 
-  at::Tensor A_scale_inverse_arg = A_scale_inverse.clone();
+  at::Tensor A_scale_inverse_arg = A_scale_inverse;
   if (A_scale_inverse.numel())
     A_scale_inverse_arg = A_scale_inverse[A_fp8_tensor];
 
-  at::Tensor B_scale_inverse_arg = B_scale_inverse.clone();
+  at::Tensor B_scale_inverse_arg = B_scale_inverse;
```
Comment (ptrendx) [inline transformer_engine/pytorch/csrc/ts_fp8_op.cpp:99]: Can't we just use the original value instead? This just makes the code harder to read for no reason.

Comment (ksivaman) [reply transformer_engine/pytorch/csrc/ts_fp8_op.cpp:99]: Sure we could, we still need to check that tensor is not empty though.

**Review by ptrendx — APPROVED** [review]
LGTM
